Userland/Libraries/LibJS/Tests/builtins/RegExp/RegExp.legacy.js

```diff
@@ -0,0 +1,160 @@
+    expect(RegExp.lastMatch).toBe('789.012');
+    expect(RegExp['$&']).toBe('789.012');
+    expect(RegExp.lastParen).toBe('012');
+    expect(RegExp['$+']).toBe('012');
+    expect(RegExp.leftContext).toBe('ghi');
+    expect(RegExp['$`']).toBe('ghi');
+    expect(RegExp.rightContext).toBe('jkl');
+    expect(RegExp["$'"]).toBe('jkl');
+
+    expect(RegExp.$1).toBe('789.012');
+    expect(RegExp.$2).toBe('789');
+    expect(RegExp.$3).toBe('012');
+    expect(RegExp.$4).toBe('');
+    expect(RegExp.$5).toBe('');
+    expect(RegExp.$6).toBe('');
+    expect(RegExp.$7).toBe('');
+    expect(RegExp.$8).toBe('');
+    expect(RegExp.$9).toBe('');
+});
+
+
+test('replace should all behave as if exec were called.', () => {
+    var re = /((\d+)\.(\d+))/;
+    var s = 'abc123.456def';
+    s.replace(re, 'whocares');
+
+    expect(RegExp.input).toBe(s);
+    expect(RegExp['$_']).toBe(s);
+    expect(RegExp.lastMatch).toBe('123.456');
+    expect(RegExp['$&']).toBe('123.456');
+    expect(RegExp.lastParen).toBe('456');
+    expect(RegExp['$+']).toBe('456');
+    expect(RegExp.leftContext).toBe('abc');
+    expect(RegExp['$`']).toBe('abc');
+    expect(RegExp.rightContext).toBe('def');
+    expect(RegExp["$'"]).toBe('def');
+
+    expect(RegExp.$1).toBe('123.456');
+    expect(RegExp.$2).toBe('123');
+    expect(RegExp.$3).toBe('456');
+    expect(RegExp.$4).toBe('');
+    expect(RegExp.$5).toBe('');
+    expect(RegExp.$6).toBe('');
+    expect(RegExp.$7).toBe('');
+    expect(RegExp.$8).toBe('');
+    expect(RegExp.$9).toBe('');
+});
+
+test('test should all behave as if exec were called.', () => {
+    var re = /((\d+)\.(\d+))/;
+    var s = 'ghi789.012jkl';
+    re.test(s);
+
+    expect(RegExp.input).toBe(s);
+    expect(RegExp['$_']).toBe(s);
+    expect(RegExp.lastMatch).toBe('789.012');
+    expect(RegExp['$&']).toBe('789.012');
+    expect(RegExp.lastParen).toBe('012');
+    expect(RegExp['$+']).toBe('012');
+    expect(RegExp.leftContext).toBe('ghi');
+    expect(RegExp['$`']).toBe('ghi');
+    expect(RegExp.rightContext).toBe('jkl');
+    expect(RegExp["$'"]).toBe('jkl');
+
+    expect(RegExp.$1).toBe('789.012');
+    expect(RegExp.$2).toBe('789');
+    expect(RegExp.$3).toBe('012');
+    expect(RegExp.$4).toBe('');
+    expect(RegExp.$5).toBe('');
+    expect(RegExp.$6).toBe('');
+    expect(RegExp.$7).toBe('');
+    expect(RegExp.$8).toBe('');
+    expect(RegExp.$9).toBe('');
+});
+
+test('replace must interleave matching and replacing when a global regexp is matched and replaced with the result of a function', () => {
+    var re = /(.)/g;
+    function f() { return RegExp.$1; };
+    expect('abcd'.replace(re, f)).toBe('dddd');
+});
+
+test('the last parenthesis didn\'t match.', () => {
+    expect(/foo(?:a(x))?/.exec("foobx")).toEqual(["foo", undefined]);
+    expect(RegExp.lastParen).toBe('');
+});
+
+test('$1 to $9', () => {
+    for (var i = 1; i <= 9; i++) {
+        var haystack = "foo";
+        var re_text = "^foo";
```

Comment:
In javascript we use camelCase, but I think this is the only variable currently not matching that.